### PR TITLE
include gain and elevation in valid beacon and witness reports

### PR DESCRIPTION
**Pull request by andymck**
proto pr: https://github.com/helium/proto/pull/328

### CODE REVIEW

File: file_store/src/iot_valid_poc.rs [hunk 1]
```diff
@@ -23,6 +23,8 @@ pub const SCALING_PRECISION: u32 = 4;
 pub struct IotValidBeaconReport {
     pub received_timestamp: DateTime<Utc>,
     pub location: Option<u64>,
+    pub gain: i32,
```
[the remark this answers is not in the document]
Comment (andymck) [reply file_store/src/iot_valid_poc.rs:26]: I thought the same originally and had the gain and elevation as an option. But it made no sense to do so as these are only contained within a valid beacon or valid witness report and as such we know we do have those values.  

The location could also be dropped as an option but i dont want to impact existing data structures ( at least not until we explicitly determine we want to do that and even then we should prob address that via a separate PR )

**Comment by vihu** [comment]
@andymck Looks like this is failing build on a missing field `nsid` on `ProtocolHttpRoamingV1`. Does this and the underlying proto pr need rebasing?

**Comment by andymck** [comment]
the fix for the missing field is part of #510

**Review by jeffgrunewald — APPROVED** [review]

**Comment by andymck** [comment]
Confirmed on a local run the elevation and gain fields are present in the iot_poc files outputted to s3:

{
      "elevation": 13,
      "gain": 23,
      "hex_scale": 702,
      "invalid_reason": 0,
      "location": "631243922681048063",
      "participant_side": 0,
      "received_timestamp": 1684414953323,
      "report": {
        "data": [
          234,
          193,
          ...
        ],
        "datarate": 3,
        "frequency": 904300000,
        "pub_key": [
          0,
          163,
          ...
        ],
        "signal": -1190,
        "signature": [
          48,
          69,
          ...
        ],
        "snr": -125,
        "timestamp": 1684414952988465822,
        "tmst": 324895366
      },
      "reward_unit": 2220,
      "status": 0
    },

**Pull request merged**